eth/catalyst/simulated_beacon.go

```diff
@@ -157,14 +159,21 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
 	}
 	payload := envelope.ExecutionPayload
 
+	var finalizedHash common.Hash
+	if payload.Number%devEpochLength == 0 {
+		finalizedHash = payload.BlockHash
+	} else {
+		finalizedHash = c.eth.BlockChain().GetBlockByNumber((payload.Number - 1) / 32 * 32).Hash()
```

Comment:
```suggestion
		finalizedHash = c.eth.BlockChain().GetBlockByNumber((payload.Number - 1) / devEpochLength * devEpochLength).Hash()
```
More correct, now that we have a constant, right?